janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphIndexTest.java

```diff
@@ -1487,6 +1488,66 @@ public void testIndexInlineProperties() throws NoSuchMethodException {
         assertEquals(city, v.value("city"));
     }
 
+    @Test
+    public void testIndexInlinePropertiesReindex() throws NoSuchMethodException, ExecutionException, InterruptedException {
+
+        clopen(option(FORCE_INDEX_USAGE), true);
+
+        PropertyKey idKey = makeKey("id", Integer.class);
+        PropertyKey nameKey = makeKey("name", String.class);
+        PropertyKey cityKey = makeKey("city", String.class);
+
+        mgmt.buildIndex("composite", Vertex.class)
+            .addKey(cityKey)
+            .buildCompositeIndex();
+
+        finishSchema();
+
+        String city = "Chicago";
+        for (int i = 0; i < 3; i++) {
+            tx.addVertex("id", i, "name", "name" + i, "city", city);
+        }
+
+        tx.commit();
+
+        tx = graph.buildTransaction()
+            .propertyPrefetching(false) //this is important
+            .start();
+
+        Method m = VertexCentricQueryBuilder.class.getSuperclass().getDeclaredMethod("constructQuery", RelationCategory.class);
+        m.setAccessible(true);
+
+        List<Vertex> vertices = tx.traversal().V().has("city", city).toList();
+        vertices.stream()
+            .map(v -> (CacheVertex) v)
+            .forEach(v -> verifyPropertyLoaded(v, "name", false, m));
+
+        tx.commit();
+
+        //Include inlined property and reindex
+        JanusGraphIndex index = mgmt.getGraphIndex("composite");
+        nameKey = mgmt.getPropertyKey("name");
+        mgmt.addInlinePropertyKey(index, nameKey);
+        ScanJobFuture scanJobFuture = mgmt.updateIndex(index, SchemaAction.REINDEX);
+        finishSchema();
+
+        while (!scanJobFuture.isDone()) {
+            Thread.sleep(1000);
+        }
```

Comment:
I'm not sure, but `finishSchema` commits current `mgmt` and opens a new one immediately. It could potentially result in an issue I guess, but that's just speculation.
What if you change this block to the following?
```
mgmt.updateIndex(index, SchemaAction.REINDEX).get();
finishSchema();
```
Can't say it will help, but it we can try.

Reply:
@porunov, thank you for looking into this.
I have added an extra call of `finishSchema()` after modifying the index structure and another one after reindexing the job. The test became stable.

Reply:
Great. Thank you!